misc/python/materialize/checks/drop_subsource.py

```diff
@@ -0,0 +1,89 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+from textwrap import dedent
+from typing import List
+
+from materialize.checks.actions import Testdrive
+from materialize.checks.checks import Check
+from materialize.checks.common import KAFKA_SCHEMA_WITH_SINGLE_STRING_FIELD
+
+
+class DropSubsource(Check):
+    def initialize(self) -> Testdrive:
+        return Testdrive(
+            dedent(
+                """
+                > CREATE SECRET pgpass AS 'postgres'
+                > CREATE CONNECTION pg TO POSTGRES (
+                    HOST 'postgres',
+                    DATABASE postgres,
+                    USER postgres,
+                    PASSWORD SECRET pgpass
+                  );
+        
+                $ postgres-execute connection=postgres://postgres:postgres@postgres
+                ALTER USER postgres WITH replication;
+                DROP SCHEMA IF EXISTS public CASCADE;
+                DROP PUBLICATION IF EXISTS mz_source;
+                CREATE SCHEMA public;
+        
+                CREATE TABLE t1 (f1 INTEGER);
+                ALTER TABLE t1 REPLICA IDENTITY FULL;
+                INSERT INTO t1 VALUES (1);
+        
+                CREATE TABLE t2 (f1 INTEGER);
+                ALTER TABLE t2 REPLICA IDENTITY FULL;
+                INSERT INTO t2 VALUES (1);
+        
+                CREATE PUBLICATION mz_source FOR ALL TABLES;
+        
+                > CREATE SOURCE mz_source
+                  FROM POSTGRES CONNECTION pg
+                  (PUBLICATION 'mz_source')
+                  FOR TABLES (t1);
+        
+                > SELECT COUNT(*) = 1 FROM t1;
+                true
+                """
+            )
+        )
+
+    def manipulate(self) -> List[Testdrive]:
+        return [
+            Testdrive(dedent(s))
+            for s in [
+                """
+                $ postgres-execute connection=postgres://postgres:postgres@postgres
+                CREATE TABLE t3 (f1 INTEGER);
+                ALTER TABLE t3 REPLICA IDENTITY FULL;
+                INSERT INTO t3 VALUES (1);
+                
+                > ALTER SOURCE mz_source
+                  ADD SUBSOURCE t3;
+                """,
+                """
+                $ postgres-execute connection=postgres://postgres:postgres@postgres
+                > ALTER SOURCE mz_source
+                  DROP SUBSOURCE t3;
+                
+                > ALTER SOURCE mz_source
+                  ADD SUBSOURCE t3;
+                """,
+            ]
+        ]
+
+    def validate(self) -> Testdrive:
+        return Testdrive(
+            dedent(
+                """
+            ! DROP SOURCE t3;
```

Comment:
Here , you need two identical error checks -- one each for the subsources created in manipulate1 and in manipulate2

Reply:
✅